extensions/sql-database-projects/src/dialogs/addDatabaseReferenceDialog.ts

```diff
@@ -261,26 +262,12 @@ export class AddDatabaseReferenceDialog {
 			this.setDefaultDatabaseValues();
 		});
 
-		// get projects in workspace
-		const workspaceFolders = vscode.workspace.workspaceFolders;
-		if (workspaceFolders?.length) {
-			let projectFiles = await utils.getSqlProjectFilesInFolder(workspaceFolders[0].uri.fsPath);
+		// get projects in workspace and filter to only sql projects
+		let projectFiles: vscode.Uri[] = (vscode.extensions.getExtension(dataworkspace.extension.name)?.exports.getProjectsInWorkspace()).filter((p: vscode.Uri) => path.parse(p.fsPath).ext === constants.sqlprojExtension);
```

Comment:
you should use the dataworkspace.IExtension here, I think you are working with 'any' type after 'exports'.